components/selection/selection-mixin.js

```diff
@@ -0,0 +1,132 @@
+/*
+export const selectionTypes = {
+	none: 'none',
+	single: 'single',
+	multiple: 'multiple'
+}
+*/
+
+export class SelectionInfo {
+
+	constructor(keys, state) {
+		if (!keys) keys = [];
+		if (!state) state = SelectionInfo.states.none;
+		this._keys = keys;
+		this._state = state;
+	}
+
+	get keys() {
+		return this._keys;
+	}
+
+	get state() {
+		return this._state;
+	}
+
+	static get states() {
+		return {
+			none: 'none',
+			some: 'some',
+			all: 'all'
+		};
+	}
+
+}
+
+export const SelectionMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			//selectionType: { type: String, attribute: 'selection-type' }
```

Comment:
Remove until we know more?

Reply:
Yeah... will remove this stuff and add back in the near future.